data-model/src/main/scala/za/co/absa/enceladus/model/properties/PropertyDefinition.scala

```diff
@@ -0,0 +1,97 @@
+                              essentiality: Essentiality = Optional(),
+                              disabled: Boolean = false,
+
+                              // VersionModel induced fields:
+                              dateCreated: ZonedDateTime = ZonedDateTime.now(),
+                              userCreated: String = null,
+
+                              lastUpdated: ZonedDateTime = ZonedDateTime.now(),
+                              userUpdated: String = null,
+
+                              dateDisabled: Option[ZonedDateTime] = None,
+                              userDisabled: Option[String] = None,
+                              parent: Option[MenasReference] = None
+                             ) extends VersionedModel with Auditable[PropertyDefinition] {
+
+  val isRequired: Boolean = essentiality == Mandatory()
+  val isOptional: Boolean = essentiality == Optional()
+
+  // VersionModel induced methods:
+  override def setVersion(value: Int): PropertyDefinition = this.copy(version = value)
+  override def setDisabled(disabled: Boolean): PropertyDefinition = this.copy(disabled = disabled)
+  override def setLastUpdated(time: ZonedDateTime): PropertyDefinition = this.copy(lastUpdated = time)
+  override def setUpdatedUser(user: String): PropertyDefinition = this.copy(userUpdated = user)
+  override def setDescription(desc: Option[String]): PropertyDefinition = this.copy(description = desc)
+  override def setDateCreated(time: ZonedDateTime): PropertyDefinition = this.copy(dateCreated = time)
+  override def setUserCreated(user: String): PropertyDefinition = this.copy(userCreated = user)
+  override def setDateDisabled(time: Option[ZonedDateTime]): PropertyDefinition = this.copy(dateDisabled = time)
+  override def setUserDisabled(user: Option[String]): PropertyDefinition = this.copy(userDisabled = user)
+  override def setParent(newParent: Option[MenasReference]): PropertyDefinition = this.copy(parent = newParent)
+
+  // Auditable induced methods:
+  override val createdMessage = AuditTrailEntry(menasRef = MenasReference(collection = None, name = name, version = version),
+    updatedBy = userUpdated, updated = lastUpdated, changes = Seq(
+      AuditTrailChange(field = "", oldValue = None, newValue = None, s"PropertyDefinition $name created.")))
+
+  override def getAuditMessages(newRecord: PropertyDefinition): AuditTrailEntry = {
+    AuditTrailEntry(menasRef = MenasReference(collection = None, name = newRecord.name, version = newRecord.version),
+      updated = newRecord.lastUpdated,
+      updatedBy = newRecord.userUpdated,
+      changes = super.getPrimitiveFieldsAudit(newRecord,
+        Seq(AuditFieldName("propertyType", "Property type"),
+          AuditFieldName("putIntoInfoFile", "Put into _INFO file"),
+          AuditFieldName("essentiality", "Essentiality"),
+          AuditFieldName("schemaVersion", "Schema Version"),
+          AuditFieldName("schedule", "Schedule"))))
+  }
```

Comment:
Honestly, I am not sure what should (not) be audited here. Which changes and which fields...?

Reply:
`Schema version` seems off, as does `schedule`. 

I would add the `descripton` and call it done

Reply:
Thanks.